packages/ui/src/app/Providers.tsx

```diff
@@ -21,32 +23,39 @@ interface Props {
 }
 
 export const Providers = ({ children }: Props) => (
-  <KeyringContextProvider>
-    <NetworkEndpointsProvider>
-      <ApiContextProvider>
-        <AccountsContextProvider>
-          <QueryNodeProvider>
-            <TransactionStatusProvider>
-              <BalancesContextProvider>
-                <MembershipContextProvider>
-                  <HashRouter>
-                    <RouteActions>
-                      <ModalContextProvider>
-                        <OnBoardingProvider>
-                          <ImageReportProvider>
-                            <GlobalStyle />
-                            {children}
-                          </ImageReportProvider>
-                        </OnBoardingProvider>
-                      </ModalContextProvider>
-                    </RouteActions>
-                  </HashRouter>
-                </MembershipContextProvider>
-              </BalancesContextProvider>
-            </TransactionStatusProvider>
-          </QueryNodeProvider>
-        </AccountsContextProvider>
-      </ApiContextProvider>
-    </NetworkEndpointsProvider>
-  </KeyringContextProvider>
+  <ErrorBoundary
+    FallbackComponent={ErrorFallback}
+    onReset={() => {
+      // reset the state of your app so the error doesn't happen again
+    }}
+  >
+    <KeyringContextProvider>
```

Comment:
Catching errors all the way up here is a good idea it will allow users to still use usersnap to report errors for crashes happening in providers.

However apart from that it still mean loosing the whole app and needing to reload the page. As described in the issue #4052 other boundaries should be added:
1. to catch page crashes. I think it should be around `<Screen />` in `App.tsx`. This way the navigation is still available and the user can simply click away from the faulty page.
2. to catch modal crashes. I think it has to be done around `<GlobalModals />` (although maybe there is a better solution). The error should show in a modal too. This way the rest of the app is still available and only the modal crashed.

Reply:
totally agreed will update PR